### DISCUSSION TITLE
Planning statuse 0%

### DISCUSSION THREAD

**Discussion by MSadriAghdam** [frePPLe/frepple#526] [Q&A]
Hello, dear community.
I was wondering if anyone had this issue with planning on frepple. 
I defined some resources and materials, operations and operation materials, etc. While I am trying to lunch the plan, I am getting nothing! The status would not increase from 0% even in hours! 
If I refresh it, It will show a (new) plan there, but my latest order would not have any plans, therefore the manufacturing order and capacity report = nothing!

Does anyone have any ideas?

![image](https://user-images.githubusercontent.com/62038461/166431852-d759de41-2de0-441e-bec5-ccc236e16684.png)

**Comment by jdetaeye** [comment]
You're not providing a lot of context of your question...

How did you install the software: ubuntu server, docker container, compiled from source code?
What version is this?
What's in the log file in your screenshot?

If you install the demo dataset, does that work correctly?

**Reply by MSadriAghdam** [reply]
Thank you for your reply @jdetaeye 

- I have it installed on Ubuntu 20.04 LTS 
- version [6.21.0](https://github.com/frePPLe/frepple/releases/tag/6.21.0))
- it is still processing (since 3 hours ago!), and I do not have a log file yet.

But here is the log file of the latest plan, which was like this, but I refreshed the page after some time!
https://docs.google.com/document/d/1w5iH028r5tlv3evF-nRF8JKKejNoFnJjdRDlTs5-JQg/edit?usp=sharing

The demo worked, I also tried the table model, and it also worked. 

Looking forward

thanks

**Comment by jdetaeye** [comment]
> it is still processing (since 3 hours ago!), and I do not have a log file yet.

By now the engine is either already crashed or in some infinite loop. I both cases I would have expected a log file to be there already.
There is a built-in protection against excessive large log files - so it could be that it already deleted that log after the engine crashed.

As a test you can try running from the command line.
Try running: "frepplectl runplan --constraints=13 --plantype=1 --env=supply"

> The demo worked, I also tried the table model, and it also worked.

Then the problem is clearly linked to your dataset. 

From your log file there are 2 data errors:
- ERROR **** Invalid hierarchy relation between "FST GERMANY" and "FST GERMANY" ****
  A location refers to itself as owner/parent.
- ERROR **** Invalid hierarchy relation between "Internal" and "Internal" ****
  A customer refers to itself as owner/parent.
My feeling is these don't look like directly causing your issue - but one never knows.

**Reply by MSadriAghdam** [reply]
I appreciate your support. I removed  Invalid hierarchy relations and tried again. Still the same issue, but I will try the whole process again to see what could be the reason. 
Thanks again

**Reply by MSadriAghdam** [reply]
I cleaned the database using "Empty the database," and I have the same issue even after doing so! It does not look right!

I mean, there is no order, material, etc., and I wanted to see if I go to "create a plan" and Launch the plan, what would happen!
![image](https://user-images.githubusercontent.com/62038461/166453105-db8f4844-b33e-4502-8c16-953a55b895ef.png)

BTW, how come I still see the plans? I removed everything!?

update: I created all necessary resources and materials and planned again from scratch; however, the status of the plan is still 0%, but it does the planning!! (weird!) and I have four days delay!!! While this is the only plan, all resources are only assigned to this plan! So It could have started the process earlier, right? But...
